lib/msal-browser/src/broker/nativeBroker/PlatformAuthDOMHandler.ts

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    Logger,
+    createAuthError,
+    AuthErrorCodes,
+    IPerformanceClient,
+    StringDict,
+} from "@azure/msal-common/browser";
+import {
+    PlatformBrokerRequest,
+    PlatformDOMTokenRequest,
+} from "./PlatformBrokerRequest.js";
+import { PlatformAuthConstants } from "../../utils/BrowserConstants.js";
+import {
+    PlatformBrokerResponse,
+    PlatformDOMTokenResponse,
+} from "./PlatformBrokerResponse.js";
+import { createNativeAuthError } from "../../error/NativeAuthError.js";
+import { IPlatformAuthHandler } from "./IPlatformAuthHandler.js";
+
+export class PlatformAuthDOMHandler implements IPlatformAuthHandler {
+    protected logger: Logger;
+    protected performanceClient: IPerformanceClient;
+    protected correlationId: string;
+    protected brokerId: string;
+    protected extensionVersion: string;
+    platformAuthType: string;
+
+    constructor(
+        logger: Logger,
+        performanceClient: IPerformanceClient,
+        correlationId: string
+    ) {
+        this.logger = logger;
+        this.performanceClient = performanceClient;
+        this.brokerId = PlatformAuthConstants.MICROSOFT_ENTRA_BROKERID;
+        this.correlationId = correlationId;
+        this.extensionVersion = "";
+        this.platformAuthType = PlatformAuthConstants.PLATFORM_DOM_PROVIDER;
+    }
+
+    static async createProvider(
+        logger: Logger,
+        performanceClient: IPerformanceClient,
+        correlationId: string
+    ): Promise<PlatformAuthDOMHandler | undefined> {
+        logger.trace("PlatformAuthDOMHandler: createProvider called");
+
+        // @ts-ignore
+        if (window.navigator?.platformAuthentication) {
+            const supportedContracts =
+                // @ts-ignore
+                await window.navigator.platformAuthentication.getSupportedContracts(
+                    PlatformAuthConstants.MICROSOFT_ENTRA_BROKERID
+                );
+            if (
+                supportedContracts?.includes(
+                    PlatformAuthConstants.PLATFORM_DOM_APIS
+                )
+            ) {
+                logger.trace("Platform auth api available in DOM");
+                return new PlatformAuthDOMHandler(
+                    logger,
+                    performanceClient,
+                    correlationId
+                );
+            }
+        }
+        return undefined;
+    }
+
+    /**
+     * Returns the Id for the broker extension this handler is communicating with
+     * @returns
+     */
+    getExtensionId(): string {
+        return this.brokerId;
```

Comment:
nit: Doesn't need to be a class property
```suggestion
        return PlatformAuthConstants.MICROSOFT_ENTRA_BROKERID;
```

Reply:
Addressed!